### ISSUE TITLE
using tls package handshake failed

### ISSUE THREAD

**Issue by winterland1989** [winterland1989/mysql-haskell#3]
Currently TLS support only works via `HsOpenSSL` package, `tls` package result in handshake failure:

```
"Handshake [ClientHello TLS12 (ClientRandom {unClientRandom = \"B\\179\\&5X\\169\\173\\DC1\\DEL\\153\\170\\196\\DC3\\210A\\179l\\209ks\\246\\160a\\SYN3\\142\\164r\\\\\\DLE\\227r\\226\"}) (Session Nothing) [107,103,57,51,158] [0] [ExtensionRaw 0 \"0003000000\",ExtensionRaw 65281 \"00\",ExtensionRaw 10 \"000400170019\",ExtensionRaw 11 \"0100\",ExtensionRaw 13 \"000c060105010401030102010202\"] Nothing]"
"Alert [(AlertLevel_Fatal,DecodeError)]"
"Alert [(AlertLevel_Fatal,HandshakeFailure)]"
"Alert [(AlertLevel_Warning,CloseNotify)]"
HandshakeFailed (Error_Protocol ("expecting server hello, got alert : [(AlertLevel_Fatal,DecodeError)]",True,HandshakeFailure))
```

If someone got the time to investigate, relevant code is in [TLS.hs](https://github.com/winterland1989/mysql-haskell/blob/master/Database/MySQL/TLS.hs).

As for now, I'll release v0.3.0 without tls package support, please test and report if any problem arises.

**Comment by winterland1989** [comment]
I believe this's a [tls issue](https://github.com/vincenthz/hs-tls/issues/154), let's find it out.

**Comment by winterland1989** [comment]
resolved in 2679ae7b1827b252d5d7726cf14a98c789da1689
